catalog/core/catalog-core-migratable/src/test/java/org/codice/ddf/catalog/migratable/impl/MigrationTaskManagerTest.java

```diff
@@ -0,0 +1,253 @@
+    private static final long TEST_GROUP_COUNT = 12;
+
+    private static final int TEST_FILE_SIZE = 3;
+
+    private static final int TEST_RESULTS_LIST_SIZE = 7;
+
+    private static final int TEST_FUTURES_LIST_SIZE = 4;
+
+    @Mock
+    private MigrationFileWriter mockFileWriter;
+
+    @Mock
+    private ListeningExecutorService mockExecutor;
+
+    @Mock
+    private FutureCallback<Void> mockFutureCallback;
+
+    @Mock
+    private ListenableFuture<Void> mockFutureInstance;
+
+    @Mock
+    private AtomicReference<Throwable> mockAtomicRef;
+
+    @Mock
+    private Throwable mockThrowable;
+
+    private InspectableMigrationTaskManager taskManager;
+
+    private CatalogMigratableConfig config;
+
+    private Path exportPath;
+
+    /**
+     * For testing, the export {@link Path} can be arbitrary. Our {@link CatalogMigratableConfig}
+     * object is real and initilized to defaults. The {@link MigrationTaskManager} is the class
+     * under test and is loaded with a REAL list of MOCK future objects, otherwise we will get
+     * a null pointer exception in the foreach loop of exportFinish().
+     * <p>
+     * The exception to this rule is the final test method where we examine exportMetacardQuery()
+     * and absolutely require the whole list to be a MOCK.
+     *
+     * @throws Exception
+     */
+    @Before
+    public void setup() throws Exception {
+        exportPath = Paths.get("folderA", "folderB", "testFile");
+        config = new CatalogMigratableConfig();
+        config.setExportPath(exportPath);
+        taskManager = new InspectableMigrationTaskManager();
+    }
+
+    @Test
+    public void testExportSetupInitGetsCalled() throws Exception {
+        taskManager.exportSetup();
+        verify(mockFileWriter).init(exportPath);
+    }
+
+    @Test
+    public void testExportFinishedGracefulShutdown() throws Exception {
+        when(mockExecutor.awaitTermination(1L, TimeUnit.MINUTES)).thenReturn(true);
+
+        taskManager.exportFinish();
+
+        verify(mockExecutor).shutdown();
+        verify(mockExecutor, never()).shutdownNow();
+    }
+
+    @Test
+    public void testExportFinishedInterruptedShutdown() throws Exception {
+        when(mockExecutor.awaitTermination(1L,
+                TimeUnit.MINUTES)).thenThrow(InterruptedException.class);
+
+        taskManager.exportFinish();
+
+        verify(mockExecutor).shutdown();
+        verify(mockExecutor).shutdownNow();
+    }
+
+    @Test
+    public void testExportFinishedTimeoutShutdown() throws Exception {
+        when(mockExecutor.awaitTermination(1L, TimeUnit.MINUTES)).thenReturn(false);
+
+        taskManager.exportFinish();
+
+        verify(mockExecutor).shutdown();
+        verify(mockExecutor).shutdownNow();
+    }
+
+    @Test(expected = MigrationException.class)
+    public void testExportFinishedErrorOccurred() throws Exception {
+        taskManager.getAtomicReference()
+                .set(mockThrowable);
+        taskManager.exportFinish();
```

Comment:
May want to assert that the `MigrationException.getCause()` returns the `mockThrowable`.